frontend/src/pages/connectionTypes/manage/advanced/FileUploadExtensionRow.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {
+  Button,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  InputGroup,
+  InputGroupItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { EXTENSION_REGEX } from '~/concepts/connectionTypes/fields/fieldUtils';
+
+type FileUploadExtensionRowProps = {
+  extension?: string;
+  isDuplicate?: boolean;
+  onChange: (newValue: string) => void;
+  onRemove?: () => void;
+  allowRemove: boolean;
+  textRef?: React.RefObject<HTMLInputElement>;
+};
+
+const FileUploadExtensionRow: React.FC<FileUploadExtensionRowProps> = ({
+  extension,
+  isDuplicate,
+  onRemove,
+  allowRemove,
+  onChange,
+  textRef,
+}) => {
+  const [isValid, setIsValid] = React.useState<boolean>(!isDuplicate);
+
+  React.useEffect(
+    () => setIsValid(extension ? !isDuplicate && EXTENSION_REGEX.test(extension) : true),
+    // only run on entry or if a duplicate, otherwise wait for blur
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [isDuplicate],
+  );
+
+  return (
+    <>
+      <InputGroup data-testid="file-upload-extension-row">
+        <InputGroupItem isFill>
+          <TextInput
+            name="file-extension"
+            data-testid="file-upload-extension-row-input"
+            type="text"
+            aria-label="allowed extension"
+            ref={textRef}
+            value={extension || ''}
+            placeholder={extension || 'Example: .json'}
```

Comment:
When `extension` is truthy, the value is displayed and not the placeholder. So this could be a static string.

```suggestion
            placeholder="Example: .json"
```